src/platform/site-wide/side-nav/components/SideNav.js

```diff
@@ -0,0 +1,128 @@
+// Dependencies
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { find, filter, get, startsWith, map, orderBy } from 'lodash';
+
+class SideNav extends Component {
+  static propTypes = {
+    navItemsLookup: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      navItemsLookup: props.navItemsLookup,
+    };
+  }
+
+  toggleItemExpanded = id => () => {
+    const { navItemsLookup } = this.state;
+
+    // Derive if the item is expanded.
+    const expanded = get(navItemsLookup, `[${id}].expanded`);
+
+    // Flip the item's expanded property.
+    this.setState({
+      navItemsLookup: {
+        ...navItemsLookup,
+        [id]: {
+          ...navItemsLookup[id],
+          expanded: !expanded,
+        },
+      },
+    });
+  };
+
+  renderChildItems = (parentID, depth) => {
+    const { toggleItemExpanded } = this;
+    const { navItemsLookup } = this.state;
+
+    // Derive the items to render.
+    const filteredNavItems = filter(
+      navItemsLookup,
+      item => item.parentID === parentID,
+    );
+
+    // Sort the items by `order`.
+    const sortedNavItems = orderBy(filteredNavItems, 'order', 'asc');
+
+    return map(sortedNavItems, (item, index) => {
+      // Derive the item properties.
+      const expanded = get(item, 'expanded');
+      const hasChildren = get(item, 'hasChildren', false);
+      const href = get(item, 'href');
+      const id = get(item, 'id');
+      const label = get(item, 'label', '');
+
+      // Derive if the nav item is selected and the selected class.
+      const isSelected = startsWith(window.location.pathname, href);
+      const selectedClass = isSelected ? 'selected' : '';
+
+      // Caclculate the indentation for the child items.
+      const indentation = depth >= 2 ? 20 * (depth - 1) : 20;
+
+      // Determine if we should show a line at the end.
+      const showLine = depth === 1 && index !== sortedNavItems.length - 1;
+
+      return (
+        <li className={`va-sidenav-level-${depth}`} key={id}>
+          <div
+            className={`va-sidenav-item-label ${selectedClass}`}
+            onClick={toggleItemExpanded(id)}
```

Comment:
Can we add the `onClick` to the `button` instead of the `div`?

Reply:
So currently how it works is:
1. If you click on the row, it toggles expand/collapse.
2. If you click on the label text, it will navigate you to its link.
3. If you click on the expand/collapse icon, it does nothing (but the event propagates to the row, so it then toggles expand/collapse).

If we add `onClick` to the button/a tag instead of the `div`, then it means when you click on the row where there isn't text or the expand/collapse icon, it will not expand/collapse. Is that desired behavior?

Reply:
Hm, then maybe we can add another button element for the the row text plus the expand/collapse icon?

Reply:
Anyway, I'm not sure; only suggesting this because a div with onclick behavior will probably be flagged during an accessibility audit, and the sidebar is historically looked at closely. Deque maintains our accessibility plugin (`axe-core`) and has this writeup about it, https://www.deque.com/blog/accessible-aria-buttons/

Reply:
I used the axe chrome extension, which identified an accessibility issue with the buttons as they were missing inner text or `aria-label`. So I added `aria-label` to them [here](https://github.com/department-of-veterans-affairs/vets-website/pull/10881/files#diff-aba68ce4f937195d55cd9399052709c4R95).